pkg/system-probe/api/client/check.go

```diff
@@ -210,6 +211,14 @@ func doReq(client *http.Client, req *http.Request, module types.ModuleName) (bod
 
 // GetCheck returns data unmarshalled from JSON to T, from the specified module at the /<module>/check endpoint.
 func GetCheck[T any](client *CheckClient, module types.ModuleName) (T, error) {
+	return GetWithBody[T](client, "/check", nil, module)
+}
+
+// GetWithBody makes a GET request to a module endpoint with an optional JSON
```

Comment:
HTTP GET with a body is an anti-pattern. Any reason you cannot make it a POST request?